services/src/main/java/org/keycloak/authentication/DefaultAuthenticationFlow.java

```diff
@@ -332,6 +374,8 @@ private Response processSingleFlowExecutionModel(AuthenticationExecutionModel mo
                         }
                         CredentialValidator<?> cv = (CredentialValidator<?>) localAuthenticator;
                         typeAuthExecMap.put(cv.getType(processor.getSession()), execution);
+                    } else {
+                        nonCredentialExecutions.add(execution);
```

Comment:
Don't we need to check if it is an instance of CredentialValidator (or something like this) ?

Reply:
no, this addition specifically adds flows for the non-credential authentication options.